projects/hello_firmware/src/main.c

```diff
@@ -0,0 +1,48 @@
+#include <stdint.h>  // for integer types
+
+#include "interrupt.h"  // interrupts are required for soft timers
+#include "log.h"
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for wait function
+
+#define COUNTER_A_TIMER 500
+#define COUNTER_B_TIMER 1000
+
+typedef struct Counter {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counter;
+
+void counter_a_callback(SoftTimerId timer_id, void *context) {
```

Comment:
For private functions like this we mark them `static` (which makes them not visible from other files) and prefix them with `prv_` (so we know it's a private function).